res/css/views/settings/tabs/user/_GeneralUserSettingsTab.pcss

```diff
@@ -14,54 +14,64 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-.mx_GeneralUserSettingsTab_changePassword .mx_Field {
-    margin-inline-end: var(--SettingsTab_fullWidthField-margin-inline-end);
+:root {
+    /* The top margin of subheadings within the `GeneralUserSettingsTab` */
+    --GeneralUserSettingsTab_subheading-margin-block-start: 24px;
 }
 
-.mx_GeneralUserSettingsTab_changePassword .mx_Field:first-child {
-    margin-top: 0;
-}
+.mx_GeneralUserSettingsTab {
+    .mx_GeneralUserSettingsTab_section--account :is(.mx_EmailAddresses, .mx_PhoneNumbers),
+    .mx_GeneralUserSettingsTab_section--discovery .mx_GeneralUserSettingsTab_section--discovery_existing,
+    .mx_GeneralUserSettingsTab_section_languageInput {
+        margin-inline-end: var(--SettingsTab_fullWidthField-margin-inline-end);
+    }
 
-/* TODO: Make this selector less painful */
-.mx_GeneralUserSettingsTab_accountSection .mx_SettingsTab_subheading:nth-child(n + 1),
-.mx_GeneralUserSettingsTab_discovery .mx_SettingsTab_subheading:nth-child(n + 2),
-.mx_SetIdServer .mx_SettingsTab_subheading {
-    margin-top: 24px;
-}
+    /* TODO: Make this selector less painful */
+    .mx_GeneralUserSettingsTab_section--account .mx_SettingsTab_subheading:nth-child(n + 1),
+    .mx_GeneralUserSettingsTab_section--discovery .mx_SettingsTab_subheading:nth-child(n + 2) {
+        margin-top: var(--GeneralUserSettingsTab_subheading-margin-block-start);
+    }
 
-.mx_GeneralUserSettingsTab_accountSection .mx_Spinner,
-.mx_GeneralUserSettingsTab_discovery .mx_Spinner {
-    /* Move the spinner to the left side of the container (default center) */
-    justify-content: left;
-}
+    :is(.mx_GeneralUserSettingsTab_section--account, .mx_GeneralUserSettingsTab_section--discovery) {
+        .mx_Spinner {
+            /* Move the spinner to the left side of the container (default center) */
+            justify-content: flex-start;
+        }
+    }
 
-.mx_GeneralUserSettingsTab_accountSection .mx_EmailAddresses,
-.mx_GeneralUserSettingsTab_accountSection .mx_PhoneNumbers,
-.mx_GeneralUserSettingsTab_discovery .mx_GeneralUserSettingsTab_discovery_existing,
-.mx_GeneralUserSettingsTab_languageInput {
-    margin-inline-end: var(--SettingsTab_fullWidthField-margin-inline-end);
-}
+    .mx_GeneralUserSettingsTab_section--account_changePassword {
+        .mx_Field {
+            margin-inline-end: var(--SettingsTab_fullWidthField-margin-inline-end);
 
-.mx_GeneralUserSettingsTab_discovery_existing {
-    display: flex;
-    align-items: center;
-    margin-bottom: 5px;
-}
```

Comment:
Following @gsouquet's comments on #10679, these were moved to the top level. It seems wrong that you should nest them now?

Reply:
[Above](https://github.com/matrix-org/matrix-react-sdk/pull/10595/files#diff-8f8bb298ce398e8724d0d757d7ae359c3ee68d3c53f5c82053bc3b05a38eefc6R24) `.mx_GeneralUserSettingsTab_section--discovery_existing` is nested, and setting this `mx_GeneralUserSettingsTab_section--discovery_existing` to a different level can lead to a regression due to the difference of specificity.

This means that `mx_GeneralUserSettingsTab` on [this line above](https://github.com/matrix-org/matrix-react-sdk/pull/10595/files#diff-8f8bb298ce398e8724d0d757d7ae359c3ee68d3c53f5c82053bc3b05a38eefc6R22) should be removed,  in order to remove the difference of specificity altogether.

I am going to push a commit to address this.

Reply:
Addressed with https://github.com/matrix-org/matrix-react-sdk/pull/10595/commits/4cb0e593ea5951fa15e3dfe9f4a35f32cbc0625f.

Reply:
And I remember that comment was directed to nesting of `mx_GeneralUserSettingsTab_section--discovery_existing_address` and `mx_GeneralUserSettingsTab_section--discovery_existing_promptText` into `mx_GeneralUserSettingsTab_section--discovery_existing`, so the nesting by this PR was not related to that comment anyway, because this PR was going to nest those selectors' declarations to `mx_GeneralUserSettingsTab`.